pyaedt/workflows/icepak/power_map_from_csv.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import csv
+import os.path
+
+import pyaedt
+from pyaedt import Icepak
+import pyaedt.workflows
+from pyaedt.workflows.misc import get_aedt_version
+from pyaedt.workflows.misc import get_arguments
+from pyaedt.workflows.misc import get_port
+from pyaedt.workflows.misc import get_process_id
+from pyaedt.workflows.misc import is_student
+
+port = get_port()
+version = get_aedt_version()
+aedt_process_id = get_process_id()
+is_student = is_student()
+
+# Extension batch arguments
+extension_arguments = {"file_path": ""}
+extension_description = "Power map from file"
+
+
+def frontend():  # pragma: no cover
+
+    import tkinter
+    from tkinter import filedialog
+    from tkinter import ttk
+
+    import PIL.Image
+    import PIL.ImageTk
+
+    app = pyaedt.Desktop(
+        new_desktop=False,
+        version=version,
+        port=port,
+        aedt_process_id=aedt_process_id,
+        student_version=is_student,
+    )
+
+    active_project = app.active_project()
+    active_design = app.active_design()
+
+    project_name = active_project.GetName()
+
+    if active_design.GetDesignType() in ["Icepak"]:
+        design_name = active_design.GetName()
+    else:  # pragma: no cover
+        app.logger.debug("Icepak project is needed.")
+        app.release_desktop(False, False)
+        raise Exception("Icepak 3D Layout project is needed.")
+
+    ipk = Icepak(project_name, design_name)
+
+    # Create UI
+    master = tkinter.Tk()
+
+    master.geometry("600x150")
+
+    master.title("Create power maps from csv file")
+
+    # Load the logo for the main window
+    icon_path = os.path.join(pyaedt.workflows.__path__[0], "images", "large", "logo.png")
```

Comment:
```suggestion
    icon_path = os.path.join(pyaedt.workflows.__path__[0], "images", "large", "power_map.png")
```
I assume this is the icon that should be used ?

Reply:
This icon is for the window frame, other toolkits use ansys icon for the window frame. As this is generic for all the toolkits. I will use the same rather than the button icon (power_map which is specific to this toolkit)

Reply:
@siva-krishnaswamy Can you let me know where the power_map icon is used ? If it isn't, can you remove it from the PR ?

Reply:
It is used in the workflows/icepak/toolkits_catalog.toml file. This image will then be used as a button icon in AEDT ribbon menu.  ![image](https://github.com/user-attachments/assets/95ce34cb-34b9-4516-9981-c61a2cc5a404)

Reply:
I didn't see it in the toolkits catalog, thanks for your answer :)